packages/fxa-settings/src/components/Settings/Modal/index.tsx

```diff
@@ -119,7 +120,11 @@ export const Modal = ({
                   <button
                     className={classNames('mx-2 flex-1', confirmBtnClassName)}
                     data-testid="modal-confirm"
-                    onClick={(event) => onConfirm()}
+                    onClick={(event) => {
+                      setIsLoading(true);
```

Comment:
Does every instance of us using a modal across the codebase close after this function completes? Asking since `setIsLoading(false)` isn't called here so I'd want to double check that doesn't cause any problems 🤔

Reply:
Good point, I'll add the loading state as an optional prop and control the loading state from the `onConfirm` function that's passed in from `UnitRowRecoveryKey`.